EIPS/eip-7160.md

```diff
@@ -37,65 +37,74 @@ The key words “MUST”, “MUST NOT”, “REQUIRED”, “SHALL”, “SHALL
+    function hasPinnedTokenURI(uint256 tokenId) external view returns (bool pinned);
 }
 ```
 
 The `TokenUriPinned` event MUST be emitted when pinning a token uri with the `pinTokenUri` function.
 
 The `TokenUriUnpinned` event MUST be emitted when unpinning a token uri with the `unpinTokenUri` function.
 
-The `tokenURI` function defined in the ERC-721 Metadata standard MUST return the pinned URI when a token has a pinned uri. The `tokenURI` fucntion MUST return a default uri when a token has an unpinned uri. Which uri is returned when unpinned is up to the developer and is not specified in this standard. This ensures backwards compatibility with existing contracts and applications that rely on the single metadata URI.
+The `tokenURI` function defined in the ERC-721 Metadata extension MUST return the pinned URI when a token has a pinned uri. 
+
+The `tokenURI` function defined in the ERC-721 Metadata extension MUST return a default uri when a token has an unpinned uri.
 
 The `supportsInterface` method MUST return `true` when called with `0x06e1bc5b`.
 
-Implementing functionality to add uris to a token MUST be implemented separately from this standard. A `MetadataUpdate` or `BatchMetadataUpdate` event SHOULD be emitted when adding a uri to a token.
+Implementing functionality to add or remove uris to a token MUST be implemented separately from this standard. It is RECOMMENDED that one of the event defined in [ERC-4906](./eip-4906.md) are emitted whenever uris are added or removed.
 
 See the [Implementation](#reference-implementation) section for an example.
 
 ## Rationale
 
-The `tokenURIs` function MUST revert if the token does not exist.
-
-The `tokenURIs` function returns both the pinned URI index and the list of all metadata URIs to provide flexibility in accessing the metadata.
+Similar terminology to [ERC-721](./eip-721.md) was used in order to keep fetching metadata familiar. The concept of pinning and unpinning metadata is introduced as it is clear that NFT owners could want to choose which piece of metadata to display. At first, we considered leaving the pinning and unpinning actions up to each developer, but realized that a standard interface for pinning and unpinning allows for dApps to easily implement universal support for multi-metadata tokens. 
 
-The pinned URI can be used as a default or primary URI for the token, while the list of metadata URIs can be used to access individual assets' metadata within the token. Marketplaces could present these as a gallery or media carousels.
+We first considered whether the `tokenURIs` function should return just a string array, but added the extra information so that you could get all info desired in one call instead of potentially three calls. The pinned URI can be used as the primary URI for the token, while the list of metadata URIs can be used to access individual assets' metadata within the token. dApps could present these as a gallery or media carousels.
```

Comment:
```suggestion
We first considered whether the `tokenURIs` function should return just a string array, but added the extra information so that you could get all info desired in one call instead of potentially three calls. The pinned URI should be used as the primary URI for the token, while the list of metadata URIs can be used to access individual assets' metadata within the token. dApps could present these as a gallery or media carousels.
```